go-controller/pkg/ovn/network_controller_policy_event_handler.go

```diff
@@ -2,16 +2,32 @@ package ovn
 
 import (
 	"fmt"
-	"reflect"
-
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/retry"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"reflect"
 
 	kapi "k8s.io/api/core/v1"
 	"k8s.io/client-go/tools/cache"
 )
 
+// getChildStopChan returns a new channel that doesn't affect parentChan, but will be closed when
+// parentChan is closed. May be used for child goroutines that may need to be stopped with the main goroutine or
+// separately.
+func getChildStopChan(parentChan <-chan struct{}) chan struct{} {
```

Comment:
A couple of suggestions here:

- Can you check if parentChan is already closed before spawning the goroutine?
- Could you rename the variables to parentStopChan and childStopChan
- Have you considered putting this in util? On one side, I wouldn't, because we should really promote the use of Contexts instead. But the ovn package is so big anyway that this just feels as an util anyway.

Reply:
> Can you check if parentChan is already closed before spawning the goroutine?

Functionally it should be the same (right?), but I am thinking if saving a goroutine on shutdown worth an extra `select` that may complicate the function, wdyt?

other comments should be addressed, thanks for suggestions!

Reply:
I checked with `WithCancel` which is a similar thing and it's doing just that (from `go/src/context/context.go`)
```
// propagateCancel arranges for child to be canceled when parent is.
func propagateCancel(parent Context, child canceler) {
...
	select {
	case <-done:
		// parent is already canceled
		child.cancel(false, parent.Err(), Cause(parent))
		return
	default:
	}
```

I think it doesn't complicate things too much and a nice thing to have just in case.

Reply:
sounds good, updated